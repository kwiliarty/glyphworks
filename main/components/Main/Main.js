import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components' 

import theme from '../../themes/glyphworks'

const StyledMain = styled.main`
  background-color: ${ props => props.backColor };
  color: ${ props => props.foreColor };
  padding: ${ props => props.theme.basePadding }; 
  flex-grow: 1;
`

const Main = props => {
  const backColor = theme.colors[ props.backColor ] || props.backColor
  const foreColor = theme.colors[ props.foreColor ] || props.foreColor

  return (
    <StyledMain { ...props } backColor={ backColor } foreColor={ foreColor }>
      { props.children }
    </StyledMain>
  )
}

Main.propTypes = {
  /** The background color, the name of a color from the theme */
  backColor: PropTypes.string,
  /** This component renders its children */
  children: PropTypes.node,
  /** The foreground color, the name of a color from the theme */
  foreColor: PropTypes.string,
}

Main.defaultProps = {
  backColor: 'parchment',
  foreColor: theme.mainTextColor,
}

export default Main
