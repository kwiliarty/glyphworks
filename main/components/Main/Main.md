The Main content.

```jsx
<Main>
  Using default colors
</Main>
```

```jsx
<Main foreColor='green' backColor='lightBrown'>
  Setting custom colors
</Main>
```
