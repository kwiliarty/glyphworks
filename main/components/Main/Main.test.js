import React from 'react'
import { render, screen } from '@testing-library/react'
import Main from './Main.js'

describe( 'The Main component', () => {

  it( 'displays with a default text and background color', () => {
    render( <Main /> )
    const main = screen.getByRole( 'main' )
    expect( main ).toMatchSnapshot()
  })

  it( 'can accept custom colors', () => {
    render(
      <Main foreColor='wheat' backColor='pink' />
    )
    const main = screen.getByRole( 'main' )
    expect( main ).toHaveStyleRule('color', 'wheat')
    expect( main ).toHaveStyleRule('background-color', 'pink')
    expect( main ).toMatchSnapshot()
  })
})
