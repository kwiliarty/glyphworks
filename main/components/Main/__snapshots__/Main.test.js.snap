// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Main component can accept custom colors 1`] = `
.c0 {
  background-color: pink;
  color: wheat;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<main
  class="c0"
/>
`;

exports[`The Main component displays with a default text and background color 1`] = `
.c0 {
  background-color: #fffae9;
  color: #625418;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<main
  class="c0"
/>
`;
